src/utils/__tests__/match/matchLogic.spec.ts

```diff
@@ -0,0 +1,784 @@
+        max_matches: 2,
+        is_available: true,
+        support_type: "psychological" as const,
+        support_expertise: "",
+        offers_online_support: true,
+        offers_libras_support: false,
+        lat: -8.062561606296475 as unknown as Decimal,
+        lng: -34.91897018325968 as unknown as Decimal,
+        city: "RECIFE",
+        state: "PE",
+        updated_at: new Date("2023-01-01"),
+        created_at: new Date("2023-01-01"),
+      },
+    ];
+
+    const msrLat = -23.45581927562912;
+    const msrLng = -45.066556703531184;
+
+    const closestVolunteer = findClosestVolunteer(
+      msrLat,
+      msrLng,
+      volunteerAvailability,
+      null
+    );
+
+    expect(closestVolunteer?.volunteer_id).toStrictEqual(1);
+  });
+
+  it("should return the closest volunteer within maxDistance when maxDistance is not null", () => {
+    const volunteerAvailability: VolunteerAvailability[] = [
+      {
+        volunteer_id: 1,
+        current_matches: 0,
+        max_matches: 2,
+        is_available: true,
+        support_type: "psychological" as const,
+        support_expertise: "",
+        offers_online_support: true,
+        offers_libras_support: false,
+        lat: -23.455917719549397 as unknown as Decimal,
+        lng: -45.06648159814469 as unknown as Decimal,
+        city: "UBATUBA",
+        state: "SP",
+        updated_at: new Date("2023-01-01"),
+        created_at: new Date("2023-01-01"),
+      },
+      {
+        volunteer_id: 2,
+        current_matches: 0,
+        max_matches: 2,
+        is_available: true,
+        support_type: "psychological" as const,
+        support_expertise: "",
+        offers_online_support: true,
+        offers_libras_support: false,
+        lat: -8.062561606296475 as unknown as Decimal,
+        lng: -34.91897018325968 as unknown as Decimal,
+        city: "RECIFE",
+        state: "PE",
+        updated_at: new Date("2023-01-01"),
+        created_at: new Date("2023-01-01"),
+      },
+      {
+        volunteer_id: 3,
+        current_matches: 0,
+        max_matches: 2,
+        is_available: true,
+        support_type: "psychological" as const,
+        support_expertise: "",
+        offers_online_support: true,
+        offers_libras_support: false,
+        lat: -23.526032327909142 as unknown as Decimal,
+        lng: -46.68335478204428 as unknown as Decimal,
+        city: "SAO PAULO",
+        state: "SP",
+        updated_at: new Date("2023-01-01"),
+        created_at: new Date("2023-01-01"),
+      },
+    ];
+
+    const msrLat = -23.556168561396397;
+    const msrLng = -46.67920269003776;
+    const maxDistance = 20;
+
+    const closestVolunteer = findClosestVolunteer(
+      msrLat,
+      msrLng,
+      volunteerAvailability,
+      maxDistance
+    );
+
+    expect(closestVolunteer?.volunteer_id).toStrictEqual(3);
+  });
+});
+
+describe("findVolunteerInTheSameCity()", () => {
+  it("return null if msr city and state are not_found", () => {
```

Comment:
```suggestion
  it("should return null if msr city and state are not_found", () => {
```

tem outros casos no arquivo tb